components/builder-dbcache/src/data_store.rs

```diff
@@ -88,6 +88,13 @@ pub trait BasicSet: Bucket {
     /// Type of objects stored inside this data set.
     type Record: Persistable;
 
+    /// Delete a record from the data set with the given ID.
```

Comment:
Think we have any need for soft/delayed deletes? I'm not sure either way--a clean delete is super clear but snapshots would be our only "undo" recourse.